### PR TITLE
Preserve scroll position in Modules

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-14346
affects: student, teacher
release note: none

Test plan:
- Install Student app and enable `student_modules` feature
- narmstrong > CS 1400 > Modules > Sequential > None submission types
- Tap kebab in nav bar and toggle "mark as done" requirement"
- Go back to the list
- The cell should update and scroll position should be preserved
- Go to Pages > Links to Stuff
- Tap link to Module 1
- Should auto-scroll to Module 1
- Collapse and expand sections to make sure that all works
- View modules in Teacher app as well to make sure it's all good there

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/pt7bf5k) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fpt7bf5k)
teacher: [Link](https://rebrand.ly/po7cg9d) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fpo7cg9d)
parent: [Link](https://rebrand.ly/vkuk69b) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fvkuk69b)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student, Teacher

[MBL-14346](https://instructure.atlassian.net/browse/MBL-14346)

<details><summary>:x: XCTest failed: CoreTests/ModuleListViewControllerTests/testViewDidLoad</summary>


    Assertion Failure: ModuleListViewControllerTests.swift:109: XCTUnwrap failed: expected non-nil value of type "UILabel"
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Modules/GetModules.swift [hunk 1]
```diff
@@ -19,31 +19,79 @@
...
+    public struct Response: Codable {
+        struct Section: Codable {
+            let module: APIModule
+            let items: [APIModuleItem]
+        }
+        let sections: [Section]
+    }
 
-    public let courseID: String
+    static let loadItemsQueue = DispatchQueue.global()
```
Comment (vanwagonet) [inline Core/Core/Modules/GetModules.swift:32]: I don't see this used anywhere.

File: Core/Core/Modules/GetModules.swift [hunk 2]
```diff
@@ -19,31 +19,79 @@
...
+        for section in response.sections {
+            let module = Module.save(section.module, forCourse: courseID, in: client)
+            for item in section.items {
+                let item = ModuleItem.save(item, forCourse: courseID, in: client)
+                module.items.append(item)
+                if let masteryPath = item.masteryPathItem {
+                    module.items.append(masteryPath)
+                }
+            }
+        }
```
Comment (vanwagonet) [inline Core/Core/Modules/GetModules.swift:95]: Having all this in the use case seems much better to me than the ModuleStore.

**Pull request merged**